### PR TITLE
add a workspace and run all the tests in the workspace

### PR DESCRIPTION
**Pull request by lifupan**
Add a worksapce and run all of the tests in
under this workspace.

Fixes:#155

Signed-off-by: fupan.lfp <fupan.lfp@antfin.com>

### CODE REVIEW

**Comment by lifupan** [comment]
Hi, @grahamwhaley 

I want to add much more CI jobs for this rust agent, and can you give me the right to do that?

**Comment by grahamwhaley** [comment]
@lifupan - we should discuss with @chavafg at least. Can you give us some details of what extra jobs you would like to add? We have limited CI resource, so like to try and make sure we are being efficient and sharing jobs where we can etc. thx! (more CI is always better ...)

**Comment by lifupan** [comment]
> @lifupan - we should discuss with @chavafg at least. Can you give us some details of what extra jobs you would like to add? We have limited CI resource, so like to try and make sure we are being efficient and sharing jobs where we can etc. thx! (more CI is always better ...)

Hi @grahamwhaley ，
I want to make it run as the same with the runtime or go agent version. 
You know, if we want to make the rust-agent as the first citizen or replace the go version in the future, the full CI testing is needed. And if we want much more contributions from the community, a full CI against the PR  is also needed as a gate keeper.

**Comment by chavafg** [comment]
@lifupan you mean add more for the different distros, archs and hypervisors? I'm ok with adding those... wdyt @grahamwhaley?

File: src/agent/Cargo.toml [hunk 1]
```diff
@@ -31,3 +31,12 @@ slog = { version = "2.5.2", features = ["dynamic-keys", "max_level_trace", "rele
 slog-scope = "4.1.2"
 # for testing
 tempfile = "3.1.0"
+
+[workspace]
+members = [
+    "oci",
+    "rustjail",
+    "netlink",
```
Comment (jodh-intel) [inline src/agent/Cargo.toml]: Nit: List not sorted alphabetically.

Comment (lifupan) [reply src/agent/Cargo.toml]: Thanks,  Done.

**Comment by lifupan** [comment]
> @lifupan you mean add more for the different distros, archs and hypervisors? I'm ok if we want to add those... wdyt @grahamwhaley?

Hi @chavafg , that's what I want.

**Comment by grahamwhaley** [comment]
@lifupan @chavafg - yes, I think we probably have to add more jobs to test the rust agent, presuming one day it will become the default, and then maybe we can start removing the old agent jobs.
So, for some time I predict, we will have an increase in CI jobs and thus cost/overheads - we probably need to run this past the @kata-containers/architecture-committee - maybe the Monday conference call is the best place to make the proposal.

**Comment by grahamwhaley** [comment]
s/Monday/Tuesday/ call!!

**Comment by lifupan** [comment]
> s/Monday/Tuesday/ call!!

Yeah, it's great.

**Pull request merged**